api/src/routes/subscription/index.ts

```diff
@@ -5,53 +5,82 @@ import { SubscriptionService } from '../../services/subscription-service';
 		}
 	};
 
-	requestSubscription = async (req: AuthenticatedRequest, res: Response, next: NextFunction): Promise<void> => {
+	requestSubscription = async (req: AuthenticatedRequest, res: Response, next: NextFunction) => {
 		try {
 			const channelAddress = _.get(req, 'params.channelAddress');
 			const { seed, accessRights } = req.body as RequestSubscriptionBody;
-			// TODO#25 validate
-			const channel = await this.subscriptionService.requestSubscription(req.user.identityId, channelAddress, accessRights, seed);
-			res.status(StatusCodes.CREATED).send(channel);
+			const identityId = req.user.identityId;
+
+			if (!identityId || !channelAddress) {
+				return res.sendStatus(StatusCodes.BAD_REQUEST);
+			}
+
+			const subscription = await this.subscriptionService.getSubscription(channelAddress, identityId);
+
+			if (subscription) {
+				return res.status(StatusCodes.BAD_REQUEST).send('subscription already requested');
+			}
+
+			const channel = await this.subscriptionService.requestSubscription(identityId, channelAddress, accessRights, seed);
+			return res.status(StatusCodes.CREATED).send(channel);
 		} catch (error) {
 			this.logger.error(error);
 			next(new Error('could not request the subscription'));
 		}
 	};
 
-	authorizeSubscription = async (req: AuthenticatedRequest, res: Response, next: NextFunction): Promise<void> => {
+	authorizeSubscription = async (req: AuthenticatedRequest, res: Response, next: NextFunction) => {
 		try {
 			const channelAddress = _.get(req, 'params.channelAddress');
 			const body = req.body as AuthorizeSubscriptionBody;
-			let { subscriptionLink } = body;
-			const { identityId } = body;
-			const authorId = req.user.identityId;
+			const { subscriptionLink, identityId } = body;
+			const authorId = req.user?.identityId;
+			let subscription: Subscription = undefined;
+
+			if (!authorId) {
+				return res.sendStatus(StatusCodes.BAD_REQUEST);
+			}
+
 			if (!subscriptionLink && identityId) {
-				const sub = await this.subscriptionService.getSubscription(channelAddress, identityId);
-				subscriptionLink = sub?.subscriptionLink;
+				subscription = await this.subscriptionService.getSubscription(channelAddress, identityId);
+			} else {
+				subscription = await this.subscriptionService.getSubscriptionByLink(subscriptionLink);
 			}
-			if (!subscriptionLink) {
-				throw new Error('no subscription link found or provided!');
+
+			if (!subscription || !subscription.subscriptionLink) {
+				throw new Error('no subscription found!');
 			}
-			const channel = await this.subscriptionService.authorizeSubscription(channelAddress, subscriptionLink, authorId);
-			res.status(StatusCodes.OK).send(channel);
+
+			if (subscription.isAuthorized) {
+				return res.status(StatusCodes.BAD_REQUEST).send('subscription already authorized');
+			}
+
+			const channel = await this.subscriptionService.authorizeSubscription(
+				channelAddress,
+				subscription.subscriptionLink,
+				subscription.publicKey,
```

Comment:
I would add optional chaining for all parameters of the subscription object, also on lines 69 and 79